### ISSUE TITLE
Error not correctly thrown when using event registration

### ISSUE THREAD

**Issue by mojo2012** [Afforess/Factorio-Stdlib#25]
Hi,

My event handler function crashes unexpected and without any error, exception or warning.
```
Event.registert(defines.events.on_chunk_generated, function(event)
  global.logger.log("on_chunk_generated start")
  local a = game.player.surface
  global.logger.log("on_chunk_generated end")
end)
```
The error I get when I use the standard event registration:

> Error while running the event handler: __mojo-exploration__/control.lua:61: attempt to index field 'player' (a nil value)

This error should also be shown on screen when using your event mechanism.
I debugged for like an hour before I came to try to change back to the original event registration.

**Comment by Afforess** [comment]
I wrote a test case to verify that the event was properly sending error messages to players, and it is. I think everything is working as intended, but the problem is that if there are no players online or capable of seeing the message, then the error is essentially hidden from the game. This is why:

```
local success, err = pcall(handler, event)
            if not success then
                -- may be nil in on_load
                if _G.game then
                    Game.print_all(err)
                else
                    -- no way to handle errors cleanly when the game is not up
                    error(err)
                end
                return
            end
```
When the game exists (and it does in your event), then it tries to run Game.print_all with the error message, but if there aren't any players to receive the message, then print_all isn't terribly useful. I think I can improve this - print_all returns the number of players notified, and if this is zero, then I'll raise the error the same way factorio does, so it isn't hidden.

**Comment by Afforess** [comment]
The fix is in Factorio Standard Library 0.5.1.

**Comment by mojo2012** [comment]
I'll try it out today, thanks.
